packages/audioplayers_android/android/src/main/kotlin/xyz/luan/audioplayers/player/WrappedPlayer.kt

```
@@ -232,22 +232,43 @@ class WrappedPlayer internal constructor(
      * Player callbacks
      */
     fun onPrepared() {
-        prepared = true
-        ref.handleDuration(this)
-        if (playing) {
-            player?.start()
-            ref.handleIsPlaying()
-        }
-        if (shouldSeekTo >= 0 && player?.isLiveStream() != true) {
-            player?.seekTo(shouldSeekTo)
+        try {
+            prepared = true
+            ref.handleDuration(this)
+            if (playing) {
+                player?.start()
+                ref.handleIsPlaying()
+            }
+            if (shouldSeekTo >= 0 && player?.isLiveStream() != true) {
+                player?.seekTo(shouldSeekTo)
+            }
+        } catch (e: Exception) {
+            ref.handleError(this, "WrappedPlayer exception while OnPrepared event: $e")
         }
     }
 
     fun onCompletion() {
-        if (releaseMode != ReleaseMode.LOOP) {
-            stop()
+        try {
+            if (releaseMode != ReleaseMode.LOOP) {
+                stop()
+            }
+            ref.handleComplete(this)
+        } catch (e: Exception) {
+            ref.handleError(this, "WrappedPlayer exception while OnCompletion event: $e")
+        }
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    fun onBuffering(percent: Int) {
+        // TODO(luan): expose this as a stream
+    }
+
+    fun onSeekComplete() {
+        try {
+            ref.handleSeekComplete(this)
+        } catch (e: Exception) {
+            ref.handleError(this, "WrappedPlayer exception while OnSeekComplete event: $e")
         }
-        ref.handleComplete(this)
     }
 
     fun onError(what: Int, extra: Int): Boolean {
```

I didn't until now realize (or rather remembered) that we already had an error stream. I thought I had killed it on v1 but apparently we have always been on an inconsistent state

---

I like it, as long as we make it throw the error by default, so we are not silently ignoring errors for users not using the stream
And then who is using the stream can customize that